docs/docs/ref-01-top-level-api.md

```diff
@@ -161,6 +161,10 @@ If the optional callback is provided, it will be executed after the component is
 > `ReactDOM.render()` does not modify the container node (only modifies the children of the container). In
 > the future, it may be possible to insert a component to an existing DOM node without overwriting
 > the existing children.
+>
+> `ReactDOM.render()` currently returns a reference to the root ReactComponent instance.  However, using this return value is legacy
```

Comment:
I don't want to use the word "deprecating" because we have a policy that we only mark something as "deprecated" when it's removal is imminent for the next release.  But I do like your additional explanation, so I'll work that into the message.